api-client/src/test/java/com/desk/java/apiclient/model/CustomerTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2016, Salesforce.com, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted provided
+ * that the following conditions are met:
+ *
+ *    Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright notice, this list of conditions and
+ *    the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ *    Neither the name of Salesforce.com, Inc. nor the names of its contributors may be used to endorse or
+ *    promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.desk.java.apiclient.model;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * <p> Unit tests for {@link Customer} </p>
+ *
+ * Created by Matt Kranzler on 6/24/16.
+ * Copyright (c) 2016 Desk.com. All rights reserved.
+ */
+public class CustomerTest {
+
+    private Customer customer;
+
+    @Before
+    public void setUp() throws Exception {
+        customer = new Customer();
+    }
+
+    @Test
+    public void getNameDoesReturnEmptyWithNoFirstOrLastName() throws Exception {
+        customer.setFirstName(null);
+        customer.setLastName(null);
+        assertEquals("", customer.getName());
+    }
+
+    @Test
+    public void getNameDoesReturnFirstName() throws Exception {
```

Comment:
`com.desk.java.apiclient.util.StringUtils#isEmpty` should perform a `trim()` on the source `String` before checking if it's empty. Currently we could have the case where `firstName=" "` and `lastName=" Test"` resulting in the following name `  Test`